k9mail/src/main/java/com/fsck/k9/controller/ProgressBodyFactory.java

```diff
@@ -0,0 +1,44 @@
+package com.fsck.k9.controller;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import com.fsck.k9.mail.DefaultBodyFactory;
+import org.apache.commons.io.output.CountingOutputStream;
+
+
+class ProgressBodyFactory extends DefaultBodyFactory {
+    private final ProgressListener progressListener;
```

Comment:
how about a WeakReference here? if the listener goes away we might as well just stop reporting to them